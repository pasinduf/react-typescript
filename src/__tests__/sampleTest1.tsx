import React from 'react';
import Sample from '../app/components/test/sample1'
import renderer from 'react-test-renderer';
import { cleanup, render, fireEvent, getByTitle } from '@testing-library/react';

test('sample renders correctly', () => {
    const tree = renderer.create(<Sample page={'https://www.youtube.com/'}>youtube</Sample>).toJSON();
    expect(tree).toMatchSnapshot();
});

// afterEach(cleanup);
// it('changes text in mouse hover', () => {
//     const { queryByLabelText, getByLabelText, getByText, getByTitle } = render(
//         <Sample page={'https://www.youtube.com/'}>youtube</Sample>,
//     );


// expect(queryByLabelText(/normal/i)).toBeTruthy();
// fireEvent.mouseEnter(getByLabelText(/normal/i));
// expect(queryByLabelText(/hovered/i)).toBeTruthy();

// fireEvent.mouseLeave(getByLabelText(/hovered/i));
// expect(queryByLabelText(/normal/i));

// const labelText = getByText(/normal/i);
// console.log(labelText);

// const title = getByText('title1')
// console.log(title);





//})





