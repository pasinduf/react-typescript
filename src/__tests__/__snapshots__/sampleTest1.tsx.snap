// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sample renders correctly 1`] = `
<div>
  <label
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <a
      className="normal"
      href="https://www.youtube.com/"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      target="_blank"
    >
      youtube
    </a>
    <br />
    <label>
      normal
    </label>
  </label>
  <title>
    title1
  </title>
</div>
`;
